[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66295
* __->__ #65344
* #65178

Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

swolchok committed:
[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet

Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

[ghstack-poisoned]

eellison reviewed (commented):
what's the perf improvement here ? for changes that make the code harder to follow and add complexity we should gate on perf improvements

swolchok commented:
> what's the perf improvement here ? for changes that make the code harder to follow and add complexity we should gate on perf improvements

it changes cached access from a copy to a borrow, which is huge. I think I can implement `mapTypeToAliasTypeSet` in terms of `mapTypeToBorrowedAliasTypeSet` though.

swolchok committed:
Update on "[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet"


Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet"


Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet"


Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet"


Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet"


Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

[ghstack-poisoned]

suo reviewed (approved):
approach looks good to me; given the increased salience of perf improvements here I'd say the minor complexity bump is worth

swolchok committed:
Update on "[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet"


Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet"


Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

[ghstack-poisoned]

swolchok committed:
Update on "[jit] Add MutableTypePtrHelper::mapTypeToBorrowedAliasTypeSet"


Callsites that know they are using a cache can borrow AliasTypeSets from the cache instead of copying them.

Differential Revision: [D31027363](https://our.internmc.facebook.com/intern/diff/D31027363/)

[ghstack-poisoned]
